### PR TITLE
Sync development-publish with publish branch

### PR DESCRIPTION
**Pull request by jirkadev**
1. I created a branch from HEAD of `develop-publish`
2. I merged `publish` branch to this feature branch
3. I carefully fixed all the conflicts, keeping all the firmware API changes
4. Let's merge this branch into the `development-publish` so it has the recent changes from `publish`

### CODE REVIEW

File: content/firmwareapi/pycom/machine/README.md [hunk 1]
```diff
@@ -45,26 +47,15 @@ Returns CPU frequency in hertz.
...
-_Note: in light sleep mode LoRa/Lte modems are stopped and have to be re-initialized after wakeup._
-
 #### machine.deepsleep(\[time\_ms\])
 
 Stops the CPU and all peripherals, including the networking interfaces (except for LTE). Execution is resumed from the main script, just as with a reset. If a value in milliseconds is given then the device will wake up after that period of time, otherwise it will remain in deep sleep until the reset button is pressed.
 
 The products with LTE connectivity (FiPy, GPy, G01), require the LTE radio to be disabled separately via the LTE class before entering deepsleep. This is required due to the LTE radio being powered independently and allowing use cases which require the system to be taken out from deepsleep by an event from the LTE network (data or SMS received for instance).
 
-#### machine.pin\_sleep\_wakeup(pins, mode, enable\_pull)
+#### machine.pin\_deepsleep\_wakeup(pins, mode, enable\_pull)
```
Comment (iwahdan88) [inline content/firmwareapi/pycom/machine/README.md:56]: The correct and updated version is machine.pin_sleep not deepsleep

File: content/firmwareapi/pycom/machine/README.md [hunk 2]
```diff
@@ -45,26 +47,15 @@ Returns CPU frequency in hertz.
...
 
-Configure pins to wake up from deep/light sleep mode. The pins which have this capability are: `P2, P3, P4, P6, P8 to P10 and P13 to P23`.
```
Comment (iwahdan88) [inline content/firmwareapi/pycom/machine/README.md:67]: deep/light is correct as this function sets wakeup source for both light and deep sleep

File: content/firmwareapi/pycom/machine/_index.md [hunk 3]
```diff
@@ -44,26 +46,15 @@ Returns CPU frequency in hertz.
 
 Gates the clock to the CPU, useful to reduce power consumption at any time during short or long periods. Peripherals continue working and execution resumes as soon as any interrupt is triggered (on many ports this includes system timer interrupt occurring at regular intervals on the order of millisecond).
 
-#### machine.sleep(\[time\_ms\], resume\_wifi\_ble)
```
Comment (iwahdan88) [inline content/firmwareapi/pycom/machine/_index.md]: Should not be removed this is a new added api

File: content/firmwareapi/pycom/machine/_index.md [hunk 4]
```diff
@@ -44,26 +46,15 @@ Returns CPU frequency in hertz.
...
-
-_Note: in light sleep mode LoRa/Lte modems are stopped and have to be re-initialized after wakeup._
-
 #### machine.deepsleep(\[time\_ms\])
 
 Stops the CPU and all peripherals, including the networking interfaces (except for LTE). Execution is resumed from the main script, just as with a reset. If a value in milliseconds is given then the device will wake up after that period of time, otherwise it will remain in deep sleep until the reset button is pressed.
 
 The products with LTE connectivity (FiPy, GPy, G01), require the LTE radio to be disabled separately via the LTE class before entering deepsleep. This is required due to the LTE radio being powered independently and allowing use cases which require the system to be taken out from deepsleep by an event from the LTE network (data or SMS received for instance).
 
-#### machine.pin\_sleep\_wakeup(pins, mode, enable\_pull)
```
Comment (iwahdan88) [inline content/firmwareapi/pycom/machine/_index.md]: Should not be removed as previous comments explains

File: content/firmwareapi/pycom/machine/rtc.md [hunk 5]
```diff
@@ -75,25 +81,8 @@ rtc.ntp_sync("pool.ntp.org") # this is an example. You can select a more specifi
 Returns `True` if the last `ntp_sync` has been completed, `False` otherwise:
 
 ```python
-rtc.synced()
```
Comment (iwahdan88) [inline content/firmwareapi/pycom/machine/rtc.md:78]: rtc.memory  is a new added API should not be removed

File: content/firmwareapi/pycom/network/bluetooth/README.md [hunk 6]
```diff
@@ -158,16 +171,18 @@ while bluetooth.isscanning():
             print(ubinascii.hexlify(mfg_data))
 ```
```
Comment (iwahdan88) [inline content/firmwareapi/pycom/network/bluetooth/README.md]: The removed line is the correct one the API wa updated to include timeout

File: content/firmwareapi/pycom/network/bluetooth/gattccharacteristic.md [hunk 7]
```diff
@@ -46,25 +48,3 @@ This method allows to register for notifications on the characteristic.
 * `handler` is the function that will be executed when the callback is triggered.
 * `arg` is the argument that gets passed to the callback. If nothing is given, the characteristic object that owns the callback will be used.
 
-#### characteristic.read\_descriptor(uuid)
```
Comment (iwahdan88) [inline content/firmwareapi/pycom/network/bluetooth/gattccharacteristic.md]: Should not be removed, new Addition

**Review by iwahdan88 — CHANGES_REQUESTED** [review]

File: content/firmwareapi/pycom/machine/README.md (hunk 1 above)
Comment (jirkadev) [reply content/firmwareapi/pycom/machine/README.md:56]: I tested doing changes in this file and this README.md is there with duplicated content from the old docs.

I noticed that now docs uses _index.md instead as "index folder file".

So this README.MD file can be removed. I saw these readme + _index files on many pleases.

But I would remove those in a separate PR with the base `publish`
Screen Shot 2019-10-18 at 17 30 25

File: content/firmwareapi/pycom/machine/_index.md (hunk 3 above)
Comment (jirikrepl) [reply content/firmwareapi/pycom/machine/_index.md]: this is the same problem as above

File: content/firmwareapi/pycom/machine/README.md (hunk 2 above)
Comment (jirkadev) [reply content/firmwareapi/pycom/machine/README.md:67]: This is also the README.md file which is not used anymore.

File: content/firmwareapi/pycom/machine/_index.md (hunk 3 above)
Comment (jirkadev) [reply content/firmwareapi/pycom/machine/_index.md]: OK I put it back

File: content/firmwareapi/pycom/machine/_index.md (hunk 4 above)
Comment (jirkadev) [reply content/firmwareapi/pycom/machine/_index.md]: ok done

File: content/firmwareapi/pycom/machine/rtc.md (hunk 5 above)
Comment (jirkadev) [reply content/firmwareapi/pycom/machine/rtc.md:78]: ok done

File: content/firmwareapi/pycom/machine/README.md (hunk 1 above)
Comment (iwahdan88) [reply content/firmwareapi/pycom/machine/README.md:56]: ok

File: content/firmwareapi/pycom/machine/README.md (hunk 2 above)
Comment (iwahdan88) [reply content/firmwareapi/pycom/machine/README.md:67]: ok

File: content/firmwareapi/pycom/network/bluetooth/README.md (hunk 6 above)
Comment (jirkadev) [reply content/firmwareapi/pycom/network/bluetooth/README.md]: I now removed this README.md file and added all the required changes to _index.md here 
https://github.com/pycom/pycom-documentation/pull/161/commits/c6112269bb99e7f241f3e11537d901974a32b93c

File: content/firmwareapi/pycom/network/bluetooth/gattccharacteristic.md (hunk 7 above)
Comment (jirkadev) [reply content/firmwareapi/pycom/network/bluetooth/gattccharacteristic.md]: ok done

File: content/firmwareapi/pycom/network/bluetooth/README.md (hunk 6 above)
Comment (iwahdan88) [reply content/firmwareapi/pycom/network/bluetooth/README.md]: - This new _index.md does not have the latest updates for BLE example extra params in `bluetooth.init` is removed , shouldn't be
- `bluetooth.set\_advertisement\_params` is completely removed, shouldn't be.
- `bluetooth.tx\_power` shouldn't be removed

I added the comments on file

**Comment by jirkadev** [comment]
hopefully, it should be OK, otherwise, tell me.

**Review by iwahdan88 — APPROVED** [review]

**Pull request merged**